text/0023-argo-rollouts-for-blue-green.md

```diff
@@ -0,0 +1,209 @@
+# Replace B/G deployment support of Helm-service with Argo Rollouts
+
+This KEP proposes to use Argo Rollouts for blue/green deployments instead of letting the helm-service generate a second deployment and service.
+
+## Motivation
+
+**Status quo:**
+For blue/green deployments, the Helm-service generates a second chart, which contains the primary deployment (i.e. stable version) and corresponding networking resources (i.e. VirtualServices and DesitnationRules).
+
+This generation only works under the following conditions: 
+- All services are automatically exposed via a VirtualService, i.e. Keptn cannot decide whether a service should be exposed or not
+- Istio is required
+
+**Problem statement:** The resulting drawbacks are: 
+- As the Helm chart provided by the user can only contain one deployment and one service, it is impossible to deploy multiple services together (i.e. release brackets)
+- Does not fully support edits in Git, e.g. regenerates VirtualServices when an artifact is promoted
+- Does not allow to route traffic other than HTTP
+- Does not allow to rollback to an old version, i.e. the Helm-service executes multiple Git-commits and Helm upgrades for each deployment
+- Does not allow to have different Helm charts per stage and service. This is a requirement in almost every production setting (i.e. dev has a different config than production).
+
+Overall, these drawbacks limit the blue/green deployment/releasing capabilities of the Helm-service.
+
+**Proposal:** This KEP proposes to completely remove the generation of the `generated-chart` in the helm-service and instead use [Argo Rollouts](https://argoproj.github.io/argo-rollouts/) for blue/green deployments as well as canary releases.
+
+The responsibilities of the new `helm-deploy-service` would be reduced to a Helm upgrade (i.e. `helm upgrade KEPTN_SERVICE_NAME CHART -n KEPTN_PROJECT-KEPTN_STAGE --create-namespace --install`).
+
+**Benefits:** By implementing this KEP, the following advantages can be achieved:
+- Support of Keptn service, which consists of multiple deployments, services, etc. (i.e. release brackets)
+- User can decide if a service is exposed or not
+- Argo Rollouts provide out-of-the-box canary support
```

Comment:
For support of Canary releases, more changes are required:
- ability to perform side by side SLO comparison
- in addition to threshold (metric's health), allow hypothesis/success criteria definition
- statistical metrics analysis to identify interesting movements
- stop and restart the experiment